modifiers.go

```diff
@@ -7,65 +7,96 @@ import (
 
 // Modifiers are used with update statements.
 
+// ModifierOp represents a modifier operand
+type ModifierOp int
+
 const (
-	modifierListPrepend = iota
-	modifierListAppend
-	modifierListSetAtIndex
-	modifierListRemove
-	modifierMapSetFields
-	modifierMapSetField
-	modifierCounterIncrement
+	// These modifier types represent the field modification operations
+	// on list/map types such as append/remove/map set to be used with
+	// UPDATE CQL statements
+	ModifierListPrepend      ModifierOp = iota // prepend to beginning of a list
+	ModifierListAppend                         // append to the end of a list
+	ModifierListSetAtIndex                     //	set a value for a specific list index
+	ModifierListRemove                         // remove an item from the list
+	ModifierMapSetFields                       // set values from the provided map
+	ModifierMapSetField                        // update a value for a specific key
+	ModifierCounterIncrement                   // increment a counter
 )
 
 type Modifier struct {
-	op   int
+	op   ModifierOp
 	args []interface{}
 }
 
+// Operation returns the operation this modifier represents
+func (m Modifier) Operation() ModifierOp {
```

Comment:
Doesn't really matter, but I'm not sure what advantage there is in having these values exposed via accessors rather than simply exporting the fields to be accessed directly like:

```go
type Modifier struct {
    Op   ModifierOp
    Args []interface{}
}
```

Reply:
Mostly because I did not want a Modifier to be settable arbitrarily rather than going through the constructor methods we support. This is mostly because we make assumptions in other places of the codebase.

Reply:
👍 Although it's worth noting that at least at the moment, I don't think this actually achieves/enforces that though:

* `Modifier` is passed by value, rather than by reference, everywhere as far as I can see. So if `Modifier` is included in structs internally, a caller modifying its `op` wouldn't modify its value stored internally.
* A caller could actually modify the internal state of `args` if they want:
    ```go
    modifier.Args()[0] = "foo"
   ```